src/3d/symbols/qgspoint3dsymbol_p.cpp

```diff
@@ -85,44 +85,124 @@ class QgsInstancedPoint3DSymbolHandler : public QgsFeature3DHandler
 
   private:
 
-    static Qt3DRender::QMaterial *material( const QgsPoint3DSymbol *symbol, const QgsMaterialContext &materialContext );
-    static Qt3DRender::QGeometryRenderer *renderer( const QgsPoint3DSymbol *symbol, const QVector<QVector3D> &positions );
-    static Qt3DQGeometry *symbolGeometry( const QgsPoint3DSymbol *symbol );
-
     //! temporary data we will pass to the tessellator
-    struct PointData
+    class PointData
```

Comment:
As this struct becomes a class, is there a solution to extract this class and its parent (`QgsInstancedPoint3DSymbolHandler`) in a header?

Reply:
I do not think so. Each handler defines its own slightly different `PointData`. There a 3 different one (instanced, model and billboard). It would cause confusion.